### PR TITLE
Enable award spending by agency api endpoints and bug fixes

### PR DESCRIPTION
**Pull request by jameslee40**
**High level description:**
Enables the award spending by agency api.
Fixes css issues with the table rows in the `Budget Categories` table and `Award Spending by Agency` table.
Adds request cancel logic.

**JIRA Ticket:**
N/A

**Mockup:**
https://bahdigital.invisionapp.com/share/53IAE9RE6UM

The following are ALL required for the PR to be merged:

Author:
- [N/A] Linked to this PR in JIRA ticket
- [N/A] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [N/A] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [N/A] Added Unit Tests for methods in Container Components, reducers, helper functions, and models `if applicable`
- [N/A] All `componentWillReceiveProps` and `componentWillMount` in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3277](https://federal-spending-transparency.atlassian.net/browse/DEV-3277)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [N/A] Design review complete `if applicable`
- [N/A] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/components/covid19/awardSpendingAgency/AwardSpendingAgency.jsx [hunk 1]
```diff
@@ -62,38 +61,51 @@ const AwardSpendingAgency = () => {
...
+                    const allAwardTypeGroups = [];
+                    params.filter.award_type_codes = allAwardTypeGroups.concat(...Object.values(awardTypeGroups));
+                } else {
+                    params.filter.award_type_codes = awardTypeGroups[awardType.internal];
+                }
+                return fetchAgencyCount(params).promise;
+            });
+            // Wait for all the requests to complete and then store the results in state
+            Promise.all(promises)
+                .then(([allRes, contractsRes, idvsRes, grantsRes, directPaymentsRes, loansRes, otherRes]) => {
```
Comment (MaxwellKendall) [inline src/js/components/covid19/awardSpendingAgency/AwardSpendingAgency.jsx:85]: This is a thing of b e a u t y 👏

File: src/_scss/pages/covid19/_budgetCategoriesTable.scss [hunk 2]
```diff
@@ -15,6 +15,58 @@
             }
         }
     }
+    
+    .usda-table {
+        margin-top: 0;
+        td {
+            button {
+                float: left;
+                margin-top: rem(5) !important;
```
Comment (MaxwellKendall) [inline src/_scss/pages/covid19/_budgetCategoriesTable.scss:24]: Why are we doing `!important` here? We shouldn't do this IMO. We should be able to override the component library styles or w/e relatively easily. Let me know if you need some help w/ this.

File: src/_scss/pages/covid19/_budgetCategoriesTable.scss [hunk 3]
```diff
@@ -15,6 +15,58 @@
...
+            }
+            &:first-child {
+                display: table-cell !important;
```
Comment (MaxwellKendall) [inline src/_scss/pages/covid19/_budgetCategoriesTable.scss:27]: Ditto

File: src/_scss/pages/covid19/_budgetCategoriesTable.scss [hunk 4]
```diff
@@ -15,6 +15,58 @@
...
+                }
+            }
+            @include span-columns(2 of 12, table);
+            text-align: right;
+    
+            &:not(:first-child) {
+                padding-right: rem(40);
+            }
+            .usda-table__child-cell-content {
+                padding: rem(10) rem(0) !important;
```
Comment (MaxwellKendall) [inline src/_scss/pages/covid19/_budgetCategoriesTable.scss:48]: ditto

File: src/_scss/pages/covid19/_budgetCategoriesTable.scss [hunk 5]
```diff
@@ -15,6 +15,61 @@
...
+                }
+        
+                .table-header__sort {
+                    margin-top: auto;
+                    margin-bottom: auto;
+                }
+        
+                &:not(:first-child) {
+                    .table-header__content {
+                        justify-content: flex-end;
```
Comment (MaxwellKendall) [inline src/_scss/pages/covid19/_budgetCategoriesTable.scss:65]:
```suggestion
                       @include justify-content(flex-end);
```

File: src/_scss/pages/covid19/awardSpendingAgency/_awardSpendingAgency.scss [hunk 6]
```diff
@@ -12,5 +12,77 @@
...
+                }
+        
+                .table-header__sort {
+                    margin-top: auto;
+                    margin-bottom: auto;
+                }
+        
+                &:not(:first-child) {
+                    .table-header__content {
+                        justify-content: flex-end;
```
Comment (MaxwellKendall) [inline src/_scss/pages/covid19/awardSpendingAgency/_awardSpendingAgency.scss:61]:
```suggestion
                       @include justify-content(flex-end);
```

File: src/_scss/pages/covid19/awardSpendingAgency/_awardSpendingAgency.scss [hunk 7]
```diff
@@ -12,5 +12,77 @@
...
+        
+                &:not(:first-child) {
+                    .table-header__content {
+                       @include justify-content(flex-end);
+                    }
+                }
+            }
+        }
+        
+        .table-wrapper {
```
Comment (ebdabbs) [inline src/_scss/pages/covid19/awardSpendingAgency/_awardSpendingAgency.scss:67]: We should be able to remove this block as of component library v 1.1.4 and #1811

File: src/js/dataMapping/covid19/awardSpendingAgency/awardSpendingAgencyTableTabs.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
+export const awardSpendingAgencyTableTabs = [
```
Comment (ebdabbs) [inline src/js/dataMapping/covid19/awardSpendingAgency/awardSpendingAgencyTableTabs.js:1]: We should be able to delete this array and use a more minimal version `awardTypeTabs` from `dataMapping/covid19/covid19`

File: src/js/dataMapping/covid19/awardSpendingAgency/awardSpendingAgencyTableTabs.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        columnName: 'Contract IDV'
+    }
+];
+
+export const awardSpendingAgencyTableColumnFieldMapping = {
+    name: 'name',
+    obligation: 'obligation',
+    outlay: 'outlay',
+    count: 'count',
+    faceValueOfLoan: 'face_value_of_loan'
```
Comment (ebdabbs) [inline src/js/dataMapping/covid19/awardSpendingAgency/awardSpendingAgencyTableTabs.js:58]: This mapping is the only one of real value in this object, potentially use [lodash snake case](https://lodash.com/docs/4.17.15#snakeCase) instead and we could delete this file

**Comment by MaxwellKendall** [comment]
@ebdabbs I unassigned myself b/c you're more familiar with these and have more change requests.

**Pull request merged**